nifi-api/src/main/java/org/apache/nifi/components/ValidationContext.java

```diff
@@ -99,4 +100,9 @@
      * @return the identifier of the ProcessGroup that the component being validated lives in
      */
     String getProcessGroupIdentifier();
+
+    /**
+     * @return the {@link NodeTypeProvider} which can be used to detect the node type of this NiFi instance.
+     */
+    NodeTypeProvider getNodeTypeProvider();
```

Comment:
Agreed. I was thinking it'd be too much. Exposing primary state was not what I wanted, though.. Processor's property descriptors and validators are often defined in static scope, and I thought if ValidationContext provides cluster or not, it's useful.

I'll remove this.